src/lib.rs

```diff
@@ -42,23 +42,30 @@ use types::*;
 
 /// A connection to gpsd.
 pub struct GpsdConnection {
+    raw_data: bool,
     inner: BufReader<TcpStream>
 }
 impl GpsdConnection {
     /// Make a new connection to a given address.
     pub fn new<A: ToSocketAddrs>(addr: A) -> GpsdResult<Self> {
         let stream = TcpStream::connect(addr)?;
         let inner = BufReader::new(stream);
-        Ok(Self { inner })
+        Ok(Self { inner, raw_data: false })
     }
     /// Enable or disable watcher mode.
-    pub fn watch(&mut self, watch: bool) -> GpsdResult<()> {
+    pub fn watch(&mut self, watch: bool, json: bool, raw: u8) -> GpsdResult<()> {
```

Comment:
Perhaps it might be better to split the RAW mode out into a new function, to make the API easier for newcomers who just want to get some JSON and might be confused by the additional parameters? I was thinking something like:

```rust
fn _watch(&mut self, watch: bool, json: bool, raw: u8) -> GpsdResult<()> {
    /* the current body of the watch() function */
}
/// Enable or disable watcher mode.
pub fn watch(&mut self, watch: bool) -> GpsdResult<()> {
    self._watch(watch, false, 0)
}
/// Enable RAW mode. In RAW mode, gpsd sends raw data from the GPS device, depending on the value of `raw`:
///
/// When this attribute is set to 1 for a channel, gpsd reports the unprocessed NMEA or
/// AIVDM data stream from whatever device is attached. Binary GPS packets are hex-dumped.
/// RTCM2 and RTCM3 packets are not dumped in raw mode. When this attribute is set to 2 for a channel that
/// processes binary data, gpsd reports the received data verbatim without hex-dumping.
pub fn watch_raw(&mut self, watch: bool, raw: u8) -> GpsdResult<()> {
    self._watch(watch, true, raw)
}
```

Feel free to yell if you have other ideas (to be honest, I've never used RAW mode, so I might be missing something here); I just think the above proposed API makes things a bit easier to understand for users.

Reply:
Sure, makes sense.